src/main/java/online/shop/controller/AddProductToShoppingCartController.java

```diff
@@ -22,7 +21,8 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
-        ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
+        Long userId = (Long) req.getSession().getAttribute("userId");
```

Comment:
Good job. "user_id" likes good to be a constant. Check, please, in all changed controllers

Reply:
it really looks slightly better when moved into a constant